cirq/sim/clifford/clifford_tableau.py

```diff
@@ -50,6 +51,26 @@ def bits(s):
             self.xs[i, i] = True
             self.zs[self.n + i, i] = True
 
+    def _json_dict_(self):
+        return protocols.obj_to_dict_helper(self, ['n', 'rs', 'xs', 'zs'])
+
+    @classmethod
+    def _from_json_dict_(cls, n, rs, xs, zs, **kwargs):
+        state = cls(n)
+        state.rs = rs.copy()
+        state.xs = xs.copy()
+        state.zs = zs.copy()
+
+        return state
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
```

Comment:
is there a reason why you use `type(self)` instead of writing out the type?

Reply:
It seems to be used elsewhere in the codebase, e.g.
https://github.com/quantumlib/Cirq/blob/740ef6c8034b25a0d2e6b763600b9f8cc5f5c899/cirq/ops/gate_operation.py#L83
or
https://github.com/quantumlib/Cirq/blob/740ef6c8034b25a0d2e6b763600b9f8cc5f5c899/cirq/value/timestamp.py#L80

I suspect it's because of inheritance, but my Python-fu is weak. I was just imitating what I had read.